style-guide.md

```diff
@@ -0,0 +1,220 @@
+
+There are a few ways to solve this:
+
+> **"He or she"**</br>
+> When the user submits a transaction, he or she will need to sign the message.
+> 
+> **Pick a gender**</br>
+> When the user submits a transaction, she will need to sign the message.
+> 
+> **Change the subject**</br>
+> When users submit a transaction, they will need to sign the message.
+
+### e.g. and i.e.
+
+E.g. means "for example" and is used to give a (usually non-exhaustive) example.
+
+> Substrate includes several modules (e.g., balances, consensus, assets) to help you construct your runtime.
+
+I.e. means "in other words" and is used for further clarification.
+
+> Rust is a statically-typed language (i.e., the variable types are known at compile time).
+
+E.g. and i.e. are abbreviations and should always have a period after each letter. Placing a comma after e.g. or i.e. is optional.
+
+## Programming Specific
+
+When you have a question that is not covered here, your first reference should be the Rust documentation. The Rust Book is also a great resource for style because it discusses several expressions in Rust, so you will almost always be able to find an example.
+
+### SRML Module Names
+
+SRML modules should not be capitalized and should not be in \`backticks\`.
+
+> The following examples show how to use the balances module in your custom module.
+
+When referring to the crate name, use \`backticks\`, e.g. `srml-balances`.
+
+### Types, Traits, Enums, and Structs
+
+When mentioning specific types, traits, enums (and variants), and structs, they should be in \`backticks\`.
+
+> The `Event` type is also generated as a simpler type definition for the `RawEvent` type.
+
+### Functions
+
+When mentioning a specific function, it should be in \`backticks\`. It should not include `()`.
+
+> You can execute these privileged functions by calling `sudo` with the sudo key account.
+
+When listing functions in a bulleted or numbered list, you do not need \`backticks\` if every item is strictly a function.
+
+> Functions in the `LockableCurrency` trait:
+> 
+> - set_lock
+> - extend_lock
+> - remove_lock
+
+### Macros
+
+When mentioning a specific macro in a sentence, it should be in \`backticks\` and include an exclamation mark.
+
+> At compile time, the `decl_event!` macro expands to generate the `RawEvent` enum for each module.
+
+When listing macros in a bulleted or numbered list, do not put the exclamation mark. You do not need \`backticks\` if every item is strictly a macro.
+
+> Common macros you will see in the SRML:
+> 
+> - decl_event
+> - decl_storage
+> - decl_module
+
+### Examples
+
+When including code examples in your documentation, they must comply with the [style guide](https://wiki.parity.io/Substrate-Style-Guide).
+
+### Links
+
+Links to other documents should be relative paths to avoid broken local paths or outdated http links.
+
+> ```
+> [enum](./enum.Call.html)
+> ```
+
+### Markdown Lists
```

Comment:
I am pretty sure this does not matter, but do we want to care about lists using `-` vs `*`

Or putting a space before a list item like:
```
 - there
 - is a
 - space
 - before
 - the `-`
```

If these have no affects on markdown, then I am fine to omit instructions here

Reply:
Thought about this, but AFAIK `-` and `*` are always rendered the same way. Can make a call and put it in if people think it's worthwhile.

Reply:
Na, it is just noise. One less thing to complain about in a review